extendr-api/src/robj.rs

```diff
@@ -1686,12 +1714,27 @@ pub struct StrIter {
     vector: SEXP,
     i: usize,
     len: usize,
+    levels: SEXP,
 }
 
 impl StrIter {
     /// Make an empty str iterator.
     pub fn new() -> Self {
-        unsafe { Self { vector: R_NilValue, i: 0, len: 0 } }
+        unsafe { Self { vector: R_NilValue, i: 0, len: 0, levels: R_NilValue } }
+    }
+}
+
+// Get a string reference from a CHARSXP
+fn str_from_strsxp<'a>(sexp: SEXP, index: isize) -> &'a str {
+    unsafe {
+        if index < 0 || index >= Rf_xlength(sexp) {
+            ""
```

Comment:
Do we have the concept of a missing string (like `NA_character_` in R) in Rust? I'm a bit uncomfortable with turning a missing value into an empty string.

Reply:
The correct way to do this is to use an `Option<&str>`. But as the index check should never fail with correctly formed factors, this would be a burden to users. Instead I chose to return something unobtrusive for broken factors.

Reply:
I understand your concern, but I think we need the `Option` anyways, to properly handle missing values (see #63). I believe the index lookup will fail for missing values, since they are represented as the negative largest integer (if I remember correctly).